ofrak_core/ofrak/core/free_space.py

```diff
@@ -370,15 +358,50 @@ async def _get_partially_overlapping_resources(
         )
 
 
+async def _find_and_delete_overlapping_children(resource: Resource, freed_range: Range):
+    # Note this filter calculation has the potential to be very expensive if, for instance,
+    # the resource is an entire program segment...
+    overlap_resources = list(
+        await resource.get_children_as_view(
+            MemoryRegion,
+            r_filter=ResourceFilter(
+                tags=(MemoryRegion,),
+                attribute_filters=(
+                    ResourceAttributeRangeFilter(MemoryRegion.VirtualAddress, max=freed_range.end),
+                ),
+            ),
+        )
+    )
+    for possible_overlapping_child in overlap_resources:
+        if (
+            possible_overlapping_child.virtual_address > freed_range.start
+            or possible_overlapping_child.end_vaddr() > freed_range.start
+        ):
+            await possible_overlapping_child.resource.delete()
+            await possible_overlapping_child.resource.save()
```

Comment:
All modified/deleted resources are saved after the modifier exits, and this is done before patches are applied, so why is this deleting every resource individually? If performance is a concern as the comment above implies, this will not help.

Reply:
While using this Partial modifier in more complex operations, not having this additional `save` was causing problems. For now, I propose allowing this workaround and we can further explore the underlying root cause of that failure, open another issue, and pull request as needed. We can link to this PR if desired.